### PR TITLE
Refine the doc for mul_op and fully connected layer.

### PR DESCRIPTION
**Pull request by lcy-seso**
fixes https://github.com/PaddlePaddle/Paddle/issues/6785 https://github.com/PaddlePaddle/Paddle/issues/6835

### CODE REVIEW

File: paddle/operators/mul_op.cc [hunk 1]
```diff
@@ -73,30 +73,42 @@ class MulOpMaker : public framework::OpProtoAndCheckerMaker {
...
-            `num_col_dims` dimensions, and the matrix's second dimension(row length)
-            will be the product of tensor's first `rank - num_col_dims` dimensions.
+        R"DOC(The mul_op can take tensors with more than two dimensions as its
+              inputs. If the input `X` is a tensor with more than two
+              dimensions, `X` will be flattened into a two-dimensional matrix
+              first. The flattening rule is: the first `num_col_dims` will be
+              flattened to form the first dimension of the final matrix (height
+              of the matrix), and the rest `rank(X) - num_col_dims` dimensions
+              are flattened to form the second dimension of the final matrix (
+              width of the matrix). As a result, height of the flattened matrix
```
Comment (abhinavarora) [inline paddle/operators/mul_op.cc]: `height` -> `the height`

File: paddle/operators/mul_op.cc [hunk 2]
```diff
@@ -73,30 +73,42 @@ class MulOpMaker : public framework::OpProtoAndCheckerMaker {
...
+              is equal to the product of `X`'s first `x_num_col_dims` dimensions'
+              sizes, and width of the flattened matrix is equal to the product
```
Comment (abhinavarora) [inline paddle/operators/mul_op.cc]: `width` -> `the width`

File: paddle/operators/mul_op.cc [hunk 3]
```diff
@@ -73,30 +73,42 @@ class MulOpMaker : public framework::OpProtoAndCheckerMaker {
...
+              of `X`'s last `rank(x) - num_col_dims` dimensions' size.
+              For example, suppose `X` is a 6-dimensional tensor with the shape
+              [2, 3, 4, 5, 6], and `x_num_col_dims` = 3. Then, the flattened
```
Comment (abhinavarora) [inline paddle/operators/mul_op.cc]: `Then` -> `Thus`

File: paddle/operators/mul_op.cc [hunk 4]
```diff
@@ -73,30 +73,42 @@ class MulOpMaker : public framework::OpProtoAndCheckerMaker {
...
         .SetDefault(1)
         .EqualGreaterThan(1);
     AddAttr<int>(
         "y_num_col_dims",
         "(int, default 1) "
-        R"DOC(mul_op can take tensors with more than two dimensions as input `Y`,
-             in that case, tensors will be reshaped to a matrix. Just like input `X`.
+        R"DOC(The mul_op can take tensors with more than two dimensions as its
+              inputs. If the input `Y` is a tensor with more than two
+              dimensions, `Y` will be flatten into a two-dimensional matrix
```
Comment (abhinavarora) [inline paddle/operators/mul_op.cc]: `flatten` -> `flattened`

File: python/paddle/v2/fluid/layers/nn.py [hunk 5]
```diff
@@ -28,31 +28,55 @@ def fc(input,
...
-    """
-    helper = LayerHelper('fc', **locals())
+       input: The input tensor(s) to the fully connected layer.
+       size: The number of output units in the fully connected layer.
+       num_flatten_dims: The fc layer can accept an input tensor with more than
+                         two dimensions. If this happens, the multidimensional
+                         tensor will first be flattened into a 2-dimensional
+                         matrix. The parameter `num_flatten_dims` determines
+                         how the input tensor is flattened: the first
+                         `num_flatten_dims` dimensions will be flatten to form
```
Comment (abhinavarora) [inline python/paddle/v2/fluid/layers/nn.py]: `flatten` -> `flattened`

File: paddle/operators/mul_op.cc [hunk 6]
```diff
@@ -73,30 +73,42 @@ class MulOpMaker : public framework::OpProtoAndCheckerMaker {
  public:
   MulOpMaker(OpProto* proto, OpAttrChecker* op_checker)
       : OpProtoAndCheckerMaker(proto, op_checker) {
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply paddle/operators/mul_op.cc]: I check the name convention in this doc https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/operators/op_documentation/name_convention.md, the name of the inputs and the output follows the name convention. I just fix the comment style suggested in this doc. If I miss something, please point it out. Thank you.

**Comment by abhinavarora** [comment]
LGTM!

**Pull request merged**